test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,347 @@
+                );
+              });
+            });
+          });
+
+          context(`when unset`, () => {
+            context(`when there is no default set`, () => {
+              it(`does not set ${mappedComponent}`, () => {
+                const options = MongoLogger.resolveOptions({}, {});
+                expect(options.componentSeverities).to.have.property(
+                  mappedComponent,
+                  SeverityLevel.OFF
+                );
+              });
+            });
+
+            context(`when there is a default set`, () => {
+              it(`sets ${mappedComponent} to the default`, () => {
+                const options = MongoLogger.resolveOptions(
+                  { MONGODB_LOG_ALL: SeverityLevel.DEBUG },
+                  {}
+                );
+                expect(options.componentSeverities).to.have.property(
+                  mappedComponent,
+                  SeverityLevel.DEBUG
+                );
+              });
+            });
+          });
+
+          it('is case insensitive', function () {
+            const options = MongoLogger.resolveOptions({ MONGODB_LOG_ALL: 'dEbUg' as any }, {});
+            expect(options.componentSeverities).to.have.property(
+              mappedComponent,
+              SeverityLevel.DEBUG
+            );
+          });
+        });
+      }
+    });
+
+    context('maxDocumentLength', function () {
+      const tests: Array<{
+        input: undefined | string;
+        expected: number;
+        context: string;
+        outcome: string;
+      }> = [
+        {
+          input: undefined,
+          expected: 1000,
+          context: 'when unset',
+          outcome: 'defaults to 1000'
+        },
+        {
+          input: '33',
+          context: 'when set to parsable uint',
+          outcome: 'sets `maxDocumentLength` to the parsed value',
+          expected: 33
+        },
+        {
+          input: '',
+          context: 'when set to an empty string',
+          outcome: 'defaults to 1000',
+          expected: 1000
+        },
+        {
+          input: 'asdf',
+          context: 'when set to a non-integer string',
+          outcome: 'defaults to 1000',
+          expected: 1000
+        }
+      ];
+
+      for (const { input, outcome, expected, context: _context } of tests) {
+        context(_context, () => {
+          it(outcome, () => {
+            const options = MongoLogger.resolveOptions(
+              { MONGODB_LOG_MAX_DOCUMENT_LENGTH: input },
+              {}
+            );
+            expect(options.maxDocumentLength).to.equal(expected);
+          });
+        });
+      }
+    });
+
+    context('logDestination', function () {
+      const stream = new Writable();
+      const tests: Array<{
+        env: 'stderr' | 'stdout' | undefined;
+        client: MongoLoggerMongoClientOptions['mongodbLogPath'] | undefined;
+        expectedLogDestination: MongoLoggerOptions['logDestination'];
+      }> = [
```

Comment:
we'll want to add the case-insensitive cases for logDestination and also the `<empty string>` cases (just "STDOUT" as a general stand in for case-insensitive is fine, but want to make sure both it and the empty string case is tested in both env and as a client option)